src/completers/yarn.zsh

```diff
@@ -1,8 +1,17 @@
 #!/usr/bin/env zsh
 
 _fzf_complete_yarn() {
-    if [[ $@ = 'yarn workspace ' ]]; then
-        _fzf_complete_yarn-workspace '' $@
+    local subcommand=${${(Q)${(z)@}}[2]}
+
+    if [[ $subcommand = 'workspace' ]]; then
+        local workspace
+        if ! workspace=$(_fzf_complete_parse_argument 3 1 $@ '') && [[ -z $workspace ]]; then
+            _fzf_complete_yarn-workspace '' $@
+            return
+        fi
+
+        local npm_directory=$(yarn workspaces --json info | jq -r ".data | fromjson | .[\"$workspace\"].location")
```

Comment:
Note: If Yarn 2 will be supported, this line become maybe simple by using `yarn workspaces list`.
https://yarnpkg.com/cli/workspaces/list

Reply:
What if `yarn workspaces` fails?

Reply:
Do not try to interpolate any string into jq query since it breaks too easily.
```suggestion
        local npm_directory=$(yarn workspaces --json info | jq --arg workspace "$workspace" -r '.data | fromjson | .[$workspace].location')
```